javatests/arcs/android/storage/service/StorageServiceNgImplTest.kt

```diff
@@ -0,0 +1,176 @@
+  }
+
+  @Test
+  fun openStorageChannel_invokesOnCreateCallback_andCreatesChannel() = scope.runBlockingTest {
+    val encodedStoreOptions = createEncodedStoreOptions(
+      RamDiskStorageKey("myCount"),
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    storageService.openStorageChannel(encodedStoreOptions, channelCallback, messageCallback)
+
+    // `waitForOnCreate` ensures the `onCreate` callback was invoked.
+    val channel = channelCallback.waitForOnCreate()
+
+    assertThat(channel).isInstanceOf(StorageChannelImpl::class.java)
+  }
+
+  @Test
+  fun openingTwoChannels_forSameStore_resultsInTwoDistinctChannels() = scope.runBlockingTest {
+    val encodedStoreOptions = createEncodedStoreOptions(
+      RamDiskStorageKey("myCount"),
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    val channelCallback2 = FakeStorageChannelCallback()
+    storageService.openStorageChannel(encodedStoreOptions, channelCallback, messageCallback)
+    storageService.openStorageChannel(encodedStoreOptions, channelCallback2, messageCallback)
+
+    val channel1 = channelCallback.waitForOnCreate()
+    val channel2 = channelCallback2.waitForOnCreate()
+    assertThat(channel2).isNotSameInstanceAs(channel1)
+  }
+
+  @Test
+  fun openStorageChannel_createsAStore() = scope.runBlockingTest {
+    val storageKey = RamDiskStorageKey("store1")
+    val encodedStoreOptions = createEncodedStoreOptions(
+      storageKey,
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    storageService.openStorageChannel(encodedStoreOptions, channelCallback, messageCallback)
+    val channel = channelCallback.waitForOnCreate() as StorageChannelImpl
+
+    assertThat(storageService.activeStorageKeys()).containsExactly(storageKey)
+    assertThat(channel.storageKey).isEqualTo(storageKey)
+  }
+
+  @Test
+  fun openingTwoChannels_forDifferentStores_createsTwoStores() = scope.runBlockingTest {
+    // Define the store options for two different stores.
+    val storageKey1 = RamDiskStorageKey("store1")
+    val encodedStoreOptions1 = createEncodedStoreOptions(
+      storageKey1,
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    val storageKey2 = RamDiskStorageKey("store2")
+    val encodedStoreOptions2 = createEncodedStoreOptions(
+      storageKey2,
+      CountType(),
+      ParcelableCrdtType.Count
+    )
+
+    // Open two storage channels for two distinct stores
+    storageService.openStorageChannel(encodedStoreOptions1, channelCallback, messageCallback)
+    val channel1 = channelCallback.waitForOnCreate() as StorageChannelImpl
+
+    val channelCallback2 = FakeStorageChannelCallback()
+    storageService.openStorageChannel(encodedStoreOptions2, channelCallback2, messageCallback)
+    val channel2 = channelCallback2.waitForOnCreate() as StorageChannelImpl
+
+    // Check two stores have been created.
+    assertThat(storageService.activeStorageKeys())
+      .containsExactlyElementsIn(listOf(storageKey1, storageKey2))
```

Comment:
Use `containsExactlyElementsIn` when you already have the list of results. If you need to create the list, you can use `containsExactly` instead: `.containsExactly(storageKey1, storageKey2)`